configure.cmake

```diff
@@ -161,6 +161,33 @@ IF(UNIX)
       SET(LIBWRAP "wrap")
     ENDIF()
   ENDIF()
+
+  SET(WITH_SYSTEMD "auto" CACHE STRING "Compile with systemd notification on ready")
+  IF(NOT ${WITH_SYSTEMD} MATCHES "[Nn][Oo]")
+    SET(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} systemd-daemon)
+    CHECK_C_SOURCE_COMPILES(
+    "
+    #include <systemd/sd-daemon.h>
+    int main()
+    {
+      sd_listen_fds(0);
+    }"
+    HAVE_SYSTEMD)
+    CHECK_INCLUDE_FILES (systemd/sd-daemon.h HAVE_SYSTEMD_SD_DAEMON_H)
+    CHECK_FUNCTION_EXISTS (sd_listen_fds HAVE_SYSTEMD_SD_LISTEN_FDS)
+    CHECK_FUNCTION_EXISTS (sd_notify HAVE_SYSTEMD_SD_NOTIFY)
+    CHECK_FUNCTION_EXISTS (sd_notifyf HAVE_SYSTEMD_SD_NOTIFYF)
+    IF(HAVE_SYSTEMD AND HAVE_SYSTEMD_SD_DAEMON_H AND HAVE_SYSTEMD_SD_LISTEN_FDS
+       AND HAVE_SYSTEMD_SD_NOTIFY AND HAVE_SYSTEMD_SD_NOTIFYF)
```

Comment:
This looks like a way too paranoid check. I'd probably only check for sd-daemon.h. But up to you: I don't see anything wrong about it.